development/tools/upload-block/upload-block.sh

```diff
@@ -0,0 +1,45 @@
+#! /usr/bin/env bash
+
+set -euo pipefail
+
+TENANT="anonymous"
+
+function main() {
+  if [ "$#" -ne 1 ]; then
+    echo "Expected exactly one argument: the path to a block directory." >/dev/stderr
+    exit 1
+  fi
+
+  local BLOCK_DIR="$1"
+
+  if [ ! -d "$BLOCK_DIR" ]; then
+    echo "'$BLOCK_DIR' is not a directory." >/dev/stderr
+    exit 1
+  fi
+
+  BLOCK_ULID="$(basename "$BLOCK_DIR")"
```

Comment:
If `BLOCK_ULID` isn't valid ulid, uploaded block will be ignored. We could check for it by calling `ulidtime` (from mimir/tools/ulidtime) and warn if it fails.

Reply:
I've added a check in ff0d022.